openfisca_core/formulas.py

```diff
@@ -925,77 +923,111 @@ def new_filled_column(__doc__ = None, __module__ = None,
+        cached_period_unit = periods.YEAR
+    else:
+        ValueError('set_input_divide_by_period can be used only for yearly or monthly variables.')
+
+    after_instant = period.start.offset(period_size, period_unit)
+
+    if behavior is None:
+        behavior = SET_INPUT_FILL
+
+    if behavior == SET_INPUT_FILL:
+        # Count the number of elementary periods to change, and the difference with what is already known.
+        remaining_array = array.copy()
+        sub_period = period.start.period(cached_period_unit)
+        sub_periods_count = 0
+        while sub_period.start < after_instant:
+            existing_array = holder.get_array(sub_period)
+            if existing_array is not None:
+                remaining_array -= existing_array
+            else:
+                sub_periods_count += 1
+            sub_period = sub_period.offset(1)
+
+        # Cache the input data
+        if sub_periods_count > 0:
+            divided_array = remaining_array / sub_periods_count
+            sub_period = period.start.period(cached_period_unit)
             while sub_period.start < after_instant:
-                existing_array = holder.get_array(sub_period)
-                if existing_array is not None:
-                    remaining_array -= existing_array
-                    sub_periods_count -= 1
+                if holder.get_array(sub_period) is None:
+                    holder.put_in_cache(divided_array, sub_period)
                 sub_period = sub_period.offset(1)
-            if sub_periods_count > 0:
-                divided_array = remaining_array / sub_periods_count
-                sub_period = period.start.period(period_unit)
-                while sub_period.start < after_instant:
-                    if holder.get_array(sub_period) is None:
-                        holder.put_in_cache(divided_array, sub_period)
-                    sub_period = sub_period.offset(1)
-        if period_unit == u'year':
-            remaining_array = array.copy()
-            month = period.start.period(u'month')
-            months_count = 12 * period_size
-            while month.start < after_instant:
-                existing_array = holder.get_array(month)
-                if existing_array is not None:
-                    remaining_array -= existing_array
-                    months_count -= 1
-                month = month.offset(1)
-            if months_count > 0:
-                divided_array = remaining_array / months_count
-                month = period.start.period(u'month')
-                while month.start < after_instant:
-                    if holder.get_array(month) is None:
-                        holder.put_in_cache(divided_array, month)
-                    month = month.offset(1)
-
-
-def set_input_neutralized(formula, period, array):
+        else:
+            pass  # raise ValueError('Values for variable {} are already set on period {}.'.format(holder.column.name, period))
```

Comment:
Either raise the error, or remove the else block.

If I understand well, we are in this case when in the input there is both the year value and the 12 months ones. Raising an error protects us from being inconsistant, but may be surprising for a user providing an overdefined but consistent input.